clients/settings/permissions/permissions.go

```diff
@@ -153,11 +154,18 @@ func (c *Client) delete(ctx context.Context, objectID string, accessorType strin
 		return api.Response{}, ErrorPermissions{Wrapped: err, Operation: DELETE}
 	}
 
-	httpResp, err := c.client.DELETE(ctx, path, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	httpResp, err := c.client.DELETE(ctx, path, getDefaultRequestOptions(adminAccess))
 
 	if err != nil {
 		return api.Response{}, ErrorPermissions{Wrapped: err, Operation: DELETE}
 	}
 
 	return api.NewResponseFromHTTPResponse(httpResp)
 }
+
+func getDefaultRequestOptions(adminAccess bool) rest.RequestOptions {
```

Comment:
`getDefault` would mean that there are other options. Wouldn't just `getRequestOptions` be enough?

Reply:
Good point. My original thought was that if there is some endpoint that needs different ones, this would be the default case, and other calls can adapt based on that. Because we don't have any other options here, `getRequestOptions` is sufficient, and we can adapt it in the future if even needed.